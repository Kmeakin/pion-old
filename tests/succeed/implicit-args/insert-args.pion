//~ run = "${PION} elab ${FILE}"

let id = fun (@A : Type) (a : A) => a;
let always = fun (@A : Type) (@B : Type) (a : A) (b : B) => a;

let _ = id false;
let _ = always (0 : Int) false;
{}
