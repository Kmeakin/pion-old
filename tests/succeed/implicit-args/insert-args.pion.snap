exit status: 0
stdout:
{
    let id : fun (@A : Type) -> A -> A = fun (@A : Type) (a : A) => a;
    let always : fun (@A : Type) (@B : Type) -> A -> B -> A = fun (@A : Type)
    (@B : Type)
    (a : A)
    (b : B) => a;
    let _ : Bool = id @Bool false;
    let _ : Int = always @Int @Bool 0 false;
    ()
} : ()

stderr:
